### PR TITLE
Per-player messages + mentioning

### PR DESCRIPTION
**Pull request by lucypoulton**
Includes mentioning (per-user or here/everyone/channel).

### CODE REVIEW

File: plugin/src/main/java/at/helpch/chatchat/channel/ChatChannel.java [hunk 1]
```diff
@@ -43,4 +42,12 @@ public String toString() {
                 ", channelPrefix='" + channelPrefix() +
                 '}';
     }
+
+    @Override
+    public Set<User> targets(final @NotNull User ignored) {
+        return plugin.usersHolder().users().stream().filter(user ->
+            user instanceof ChatUser &&
+                    ((ChatUser) user).player().hasPermission(ChannelUtils.SEE_CHANNEL_PERMISSION + name())
```
Comment (lucypoulton) [inline plugin/src/main/java/at/helpch/chatchat/channel/ChatChannel.java:50]: console

Comment (BlitzOffline) [reply plugin/src/main/java/at/helpch/chatchat/channel/ChatChannel.java:50]: Yeah. do something like `!(user instanceof ChatUser) || ((ChatUser) user).player().hasPermission(ChannelUtils.USE_CHANNEL_PERMISSION + name())`

File: plugin/src/main/java/at/helpch/chatchat/util/MessageProcessor.java [hunk 2]
```diff
@@ -42,33 +42,47 @@ public static void process(
...
+                final var name = ((ChatUser) target).player().getName();
+                userSpecificComponent = component.replaceText(builder ->
+                        builder.matchLiteral(mentionPrefix + name)
+                                .replacement((result, ignored) -> {
+                                    target.playSound(plugin.configManager().settings().mentionSound());
+                                    return Component.text(result.group()).color(NamedTextColor.YELLOW);
+                                })
+                );
+            }
+            target.sendMessage(user, userSpecificComponent);
```
Comment (BlitzOffline) [inline plugin/src/main/java/at/helpch/chatchat/util/MessageProcessor.java:81]: Shouldn't we still have a custom sendMessage method in the channel? That's what I was struggling with. Putting the sendMessage method in channels.

Comment (BlitzOffline) [reply plugin/src/main/java/at/helpch/chatchat/util/MessageProcessor.java:81]: Pretty sure that was the reason to move to adventure. bcz it was making it easier. I mean we can make it not adventure but still.

Comment (lucypoulton) [reply plugin/src/main/java/at/helpch/chatchat/util/MessageProcessor.java]: not sure if it's really appropriate to be honest, given that each channel is implemented differently

Comment (BlitzOffline) [reply plugin/src/main/java/at/helpch/chatchat/util/MessageProcessor.java]: Well exactly. So we can implement a different sendmessage per channel.

**Review by BlitzOffline — APPROVED** [review]

**Pull request merged**